client/src/components/topics-search.jsx

```diff
@@ -0,0 +1,196 @@
+import { navigate, Link } from '@reach/router';
+import { Dropdown, DropdownButton } from 'react-bootstrap';
+
+class TopicSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      _id: '',
+      name: '',
+      searchTerm: '',
+      searchParameter: '',
+      dropDownValue: 'Topic',
+      searchComplete: false,
+      data: [],
+    };
+  }
+
+  async componentDidMount() {
+    await ApiService.get('topics').then(
+      (res) => {
+        this.setState({
+          data: res.data,
+          loaded: true,
+        });
+      },
+      (error) => {
+        console.log('Your request to load topics list has failed because: ', error);
+        this.setState({ redirectTo: true });
+      }
+    );
+  }
+
+  handleInputChange = (event) => {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  handleChange = (event) => {
+    let target = event.target;
+    let name = target.name;
+    this.setState({
+      [name]: event.target.value,
+    });
+  };
+
+  handleCancel = (event) => {
+    event.preventDefault();
+    navigate('/topics-display');
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    let parameter = this.state.searchParameter;
+    let topic = this.state.dropDownValue;
+    let query = 'topics/searchTopics/' + topic + '/' + parameter;
+    console.log(query);
+    if (parameter.length === 0) {
+      alert('Please enter a search term');
+      navigate('/topics-search');
+    } else {
+      ApiService.get(query)
+        .then((response) => {
+          if (response.data === null) {
+            alert('Nothing found. Please enter it into the database.');
+            navigate('/topics-entry');
+          } else {
+            this.setState({
+              data: response.data,
+              searchComplete: true,
+            });
+          }
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    }
+  };
+
+  handleNewSearch = (event) => {
+    ApiService.get('topics').then((res) => {
+      this.setState({
+        data: res.data,
+        searchComplete: false,
+      });
+    });
+  };
+
+  changeValue = (event) => {
+    this.setState({ dropDownValue: event.currentTarget.textContent });
+    console.log(this.state.dropDownValue);
+  };
+
+  addMyTopics = (event) => {
+    event.preventDefault();
+  };
+
+  addMySearchTerms = (event) => {
+    event.preventDefault();
+  };
+
+  render() {
+    if (!this.state.searchComplete) {
+      return (
+        <div>
+          <h4>Search Topics</h4>
+          <div className="container-fluid">
+            <DropdownButton id="dropdown-item-button" title={this.state.dropDownValue}>
+              {this.state.data.map((topic, index) => (
+                <Dropdown.Item key={index}>
+                  <div onClick={this.changeValue}>{topic.name}</div>
+                </Dropdown.Item>
+              ))}
+            </DropdownButton>
```

Comment:
React Bootstrap... here we go!! 🚀

Reply:
I was pretty pleased with this and a bit disappointed when I got rid of the topic search functionality (that's why I didn't delete the component...). But it quickly became unnecessarily complicated